examples/training/classification/densenet/imagenet/basic_training.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+
+import tensorflow as tf
+from absl import flags
+from keras.callbacks import BackupAndRestore
+from keras.callbacks import EarlyStopping
+from keras.callbacks import ModelCheckpoint
+from keras.optimizers import Adam
+from tensorflow.keras.optimizers.schedules import PolynomialDecay
+
+import keras_cv
+from keras_cv.models import DenseNet121
+
+"""
+Title: Training a DenseNet for Imagenet Classification with KerasCV
+Author: [ianjjohnson](https://github.com/ianjjohnson)
+Date created: 2022/07/25
+Last modified: 2022/07/25
+Description: Use KerasCV to train a DenseNet using modern best practices for image classification
+"""
+
+"""
+## Overview
+KerasCV makes training state-of-the-art classification models easy by providing implementations of modern models, preprocessing techniques, and layers.
+In this tutorial, we walk through training a DenseNet model against the Imagenet dataset using Keras and KerasCV.
+This tutorial requires you to have KerasCV installed:
+```shell
+pip install keras-cv
+```
+"""
+
+"""
+## Setup and flags
+
+"""
+
+flags.DEFINE_string("imagenet_path", None, "Directory from which to load Imagenet.")
+flags.DEFINE_string(
+    "backup_path", None, "Directory which will be used for training backups"
+)
+flags.DEFINE_string(
+    "weights_path", None, "Directory which will be used to store weight checkpoints."
+)
+
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
```

Comment:
didn't know you could do this.  no action needed - but very nice to making CLI flags work with a scripted approach instead of having to define a main method.

Reply:
Yeah this is pretty nifty. I really wanted to avoid having the main method approach for these flags.